### PR TITLE
smtp: add tests for issue 7126

### PR DESCRIPTION
**Pull request by jlucovsky**
Ensure the SMTP applayer parser doesn't generate an error message while parsing the SMTP frames.

Suricata PR: https://github.com/OISF/suricata/pull/11630

## Ticket

If your pull request is related to a Suricata ticket, please provide
the full URL to the ticket here so this pull request can monitor
changes to the ticket status:

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/7126

### CODE REVIEW

**Comment by catenacyber** [comment]
Would you rebase and force-push this SV PR to get fresh CI ?

**Comment by jlucovsky** [comment]
> Would you rebase and force-push this SV PR to get fresh CI ?

Done.

File: tests/bug-7126/input.rules [hunk 1]
```diff
@@ -0,0 +1 @@
+alert ip any any -> any any (msg:"Sample rule"; flow: established, to_server; file_data; content:"no-match"; sid:1;)
```
Comment (catenacyber) [inline tests/bug-7126/input.rules:1]: Do we need a rule for this test ?

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #2053, thanks!

File: tests/bug-7126/input.rules (hunk 1 above)
Comment (catenacyber) [reply tests/bug-7126/input.rules:1]: @jlucovsky what was your intention here ?

Comment (jlucovsky) [reply tests/bug-7126/input.rules:1]: The goal was to have the BDAT section of the pcap parsed. This rule caused that to happen and hence, demonstrate the issue and the fix.

Comment (catenacyber) [reply tests/bug-7126/input.rules:1]: > The goal was to have the BDAT section of the pcap parsed.

Why is the BDAT not parsed if we do not have any rule ? 🤔

**Pull request closed without merging**